modules/osd-persistent-storage-csi-efs-sts.adoc

```diff
@@ -124,27 +125,23 @@ Save the output. You will use it in the next steps.
 +
 [source,terminal]
 ----
-$ aws iam create-policy \
+POLICY=$(aws iam create-policy \
   --policy-name "<your_rosa_cluster_name>-rosa-efs-csi" \
   --policy-document file://<your_policy_file_name>.json \
-  --query 'Policy.Arn' --output text) || \
-  POLICY=$(aws iam list-policies \
-  --query 'Policies[?PolicyName==`rosa-efs-csi`].Arn' \
-  --output text
+  --query 'Policy.Arn' --output text)
+echo $POLICY
 ----
 +
-Save the output. You will use it in the next steps.
 
 .. Attach the IAM policy to the IAM role:
 +
 [source,terminal]
 ----
 $ aws iam attach-role-policy \
      --role-name "<your_rosa_cluster_name>-aws-efs-csi-operator" \
```

Comment:
@ropatil010 Hi, does new Line 115 get the exact role-name, if yes, I think we can make it simple as below (if I understand correctly)
```suggestion
     --role-name $ROLE \
```

Reply:
HI @duanwei33, 
I guess no, the $ROLE which we get in Line:115 is not the same with --role-name $ROLE using in Line no: 141.

ROLE=$(aws iam create-role \
  --role-name "<your_rosa_cluster_name>-aws-efs-csi-operator" \ 
  
  aws iam attach-role-policy \
     --role-name "<your_rosa_cluster_name>-aws-efs-csi-operator" \

This $ROLE mentioned above we need to use in secret.yaml
stringData:
```suggestion
stringData:
  credentials: |-
    [default]
    role_arn = $ROLE
    web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
```

Reply:
Thanks, okay, so if I just need the role-name like "<your_rosa_cluster_name>-aws-efs-csi-operator", I need the cmd like:
```
ROLE_NAME=$(aws iam create-role \
  --role-name "<your_cluster_name>-aws-efs-csi-operator" \
  --assume-role-policy-document file://<your_trust_file_name>.json \
  --query "Role.RoleName" --output text)
```

Thanks to point it out. Okay let's keep this.

Reply:
Hi @smunje1, Can we make it more look better by adding like this instead of ROLE to ROLE_ARN? 

```suggestion
ROLE_ARN=$(aws iam create-role \
  --role-name "<your_rosa_cluster_name>-aws-efs-csi-operator" \
  --assume-role-policy-document file://<your_trust_file_name>.json \
  --query "Role.RoleName" --output text)
 echo $ROLE_ARN 
```
This $ROLE_ARN we are using in secret.yaml file as mentioned in above comment.

Reply:
Updated Line 141 as suggested in [above comment](https://github.com/openshift/openshift-docs/pull/64612#discussion_r1324162877).